frontend/views/Home/Admins.tsx

```diff
@@ -21,61 +51,165 @@ type AdminProps = {
               right={() =>
                 props.token !== "" && (
                   <View style={{ flexDirection: "row" }}>
-                    <IconButton icon="pencil" onPress={() => onEdit(item)} />
                     <IconButton
-                      icon="delete"
-                      onPress={() => onDelete(item.email)}
+                      icon="pencil"
+                      onPress={() => handleEditClick(item)}
                     />
+                    {superUserDeleteCheck(item)}
                   </View>
                 )
               }
             />
-          );
-        }}
-      />
-    </View>
+            <Portal>
+              {deleteModal && (
+                <Modal
+                  visible={deleteModal}
+                  contentContainerStyle={
+                    theme === "Dark" ? styles.modalDark : styles.modalLight
+                  }
+                  onDismiss={() => setDeleteModal(false)}
+                >
+                  {/*TODO: update style for text */}
+                  <Text>
+                    Are you sure you want to delete {selectedItem.email}?
+                  </Text>
+                  <Button
+                    mode="contained"
+                    onPress={() => {
+                      onDelete();
+                      setDeleteModal(false);
+                    }}
+                    style={{ backgroundColor: "red" }}
+                  >
+                    Delete
+                  </Button>
+                </Modal>
+              )}
+            </Portal>
+            <Portal>
+              {editModal && (
+                <Modal
+                  visible={editModal}
+                  contentContainerStyle={
+                    theme === "Dark" ? styles.modalDark : styles.modalLight
+                  }
+                  onDismiss={() => setEditModal(false)}
+                >
+                  {/*TODO: update style for text */}
+                  <Text>Edit admin Account</Text>
+                  <TextInput
+                    mode="outlined"
+                    left={<TextInput.Icon name="email" />}
+                    label="email"
+                    value={values.email}
+                    disabled={true}
+                  />
+                  <TextInput
+                    autoFocus={true}
+                    textContentType="name"
+                    mode="outlined"
+                    left={<TextInput.Icon name="account-badge" />}
+                    error={errors.name && touched.name}
+                    label="name"
+                    value={values.name}
+                    ref={name}
+                    onBlur={handleBlur("name")}
+                    onChangeText={handleChange("name")}
+                  />
+                  <div style={{ margin: "5px" }} />
+                  <Button mode="contained" onPress={handleSubmit}>
+                    Save
+                  </Button>
+                </Modal>
```

Comment:
Probably worth moving these Modals into their own component, they seem complicated enough. Same applies to the other Modal as well.

Reply:
They take up a lot of lines but there isn't anything happening either. If you really think they should be in separate files I can do it, but I don't thing it is necessary since it doesn't make the logic easier to follow.

Reply:
I disagree as this Modal is literally a mini Formik form that has its own operation and inputs, but sure.

I should probably move the Culture FABs into the Culture view as a whole since they don't add much and them being separate complicates things a bit.